pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCache.java

```diff
@@ -142,24 +153,38 @@ public String getActualTableName(String caseInsensitiveTableName) {
    * Returns the table config for the given table, or {@code null} if it does not exist.
    */
   @Nullable
+  @Override
   public TableConfig getTableConfig(String tableNameWithType) {
     return _tableConfigMap.get(tableNameWithType);
   }
 
+  @Override
+  public List<TableConfig> registerTableConfigChangeListener(TableConfigChangeListener tableConfigChangeListener) {
+    _tableConfigChangeListeners.add(tableConfigChangeListener);
+    return Lists.newArrayList(_tableConfigMap.values());
+  }
+
   /**
    * Returns the schema for the given table, or {@code null} if it does not exist.
    */
   @Nullable
+  @Override
   public Schema getSchema(String rawTableName) {
     String schemaName = _schemaNameMap.getOrDefault(rawTableName, rawTableName);
     SchemaInfo schemaInfo = _schemaInfoMap.get(schemaName);
     return schemaInfo != null ? schemaInfo._schema : null;
   }
 
+  @Override
+  public List<Schema> registerSchemaChangeListener(SchemaChangeListener schemaChangeListener) {
```

Comment:
should be synchronized so that we don't collide with notification that may come in another thread.